src/realm/object-store/c_api/types.hpp

```diff
@@ -616,6 +616,61 @@ struct realm_sync_session : realm::c_api::WrapC, std::shared_ptr<realm::SyncSess
     }
 };
 
+struct realm_flx_sync_subscription : realm::c_api::WrapC, realm::sync::Subscription {
+    realm_flx_sync_subscription(realm::sync::Subscription&& subscription)
+        : realm::sync::Subscription(std::move(subscription))
+    {
+    }
+};
+
+struct realm_flx_sync_subscription_set : realm::c_api::WrapC, realm::sync::SubscriptionSet {
+    realm_flx_sync_subscription_set(realm::sync::SubscriptionSet&& subscription_set)
+        : realm::sync::SubscriptionSet(std::move(subscription_set))
+    {
+    }
+};
+
+struct realm_flx_sync_mutable_subscription_set : realm::c_api::WrapC, realm::sync::MutableSubscriptionSet {
+    realm_flx_sync_mutable_subscription_set(realm::sync::MutableSubscriptionSet&& mutable_subscription_set)
+        : realm::sync::MutableSubscriptionSet(std::move(mutable_subscription_set))
+    {
+    }
+};
+
+struct realm_flx_sync_subscription_desc : realm::c_api::WrapC {
+    realm_flx_sync_subscription_desc(size_t index, bool success)
+        : m_index(index)
+        , m_success(success)
+    {
+    }
+
+    size_t get_index() const
+    {
+        return m_index;
+    }
+
+    bool success() const
```

Comment:
i think this should be called `inserted` since i was trying to match the return value of this https://en.cppreference.com/w/cpp/container/unordered_set/insert when I was designing this API.